### PR TITLE
CORTX-29834: Remove dependency from key storage, instead use cvg

### PR DESCRIPTION
**Pull request by atulsdeshmukh2312**
Signed-off-by: Atul Deshmukh <atul.deshmukh@seagate.com>

# Problem Statement
- Since 'storage' has no item other than "cvg', the hierarchy 'storage'>'cvg' can be merged to 'cvg' in GConf and changes need to be aligned accordingly wherever it is being used.
1: Current Structure:
storage:
      cvg:
      - devices:
          data:
          - /dev/sdc
          - /dev/sdd
          metadata:
          - /dev/sdb
        name: cvg-01
        type: ios
      - devices:
          data:
          - /dev/sdf
          - /dev/sdg
          metadata:
          - /dev/sde
        name: cvg-02
        type: ios
      num_cvg: 2

2: New Structure
 cvg:
    - name: cvg-02
      type: ios
      devices:
        metadata:
        - /dev/sdb
        data:
        - /dev/sdc
    - name: cvg-01
      type: ios
      devices:
        metadata:
        - /dev/sde
        data:
        - /dev/sdf
    num_cvg: 2.            
      

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 1]
```diff
@@ -872,16 +867,6 @@ def validate_storage_schema(storage):
                 for i in range(sz):
                     check_type(val[i], str, f"data_devices[{i}]")
 
-def get_cvg_cnt_and_cvg_k8(self):
```
Comment (atulsdeshmukh2312) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:875]: This function is not call from anywhere (dead code) so removing it.

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 2]
```diff
@@ -872,16 +867,6 @@ def validate_storage_schema(storage):
...
-    try:
-        cvg = self.storage['cvg']
-        cvg_cnt = len(cvg)
-    except:
-        raise MotrError(errno.EINVAL, "cvg not found\n")
-    # Check if cvg type is list
-    check_type(cvg, list, "cvg")
-    return cvg_cnt, cvg
-
 def align_val(val, size):
```
Comment (atulsdeshmukh2312) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:870]: Function 'get_cvg_cnt_and_cvg_k8' is not called from anywhere (dead code), so removing it.

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 3]
```diff
@@ -872,16 +867,6 @@ def validate_storage_schema(storage):
...
 
-def get_cvg_cnt_and_cvg_k8(self):
-    try:
-        cvg = self.storage['cvg']
-        cvg_cnt = len(cvg)
-    except:
-        raise MotrError(errno.EINVAL, "cvg not found\n")
-    # Check if cvg type is list
-    check_type(cvg, list, "cvg")
-    return cvg_cnt, cvg
```
Comment (atulsdeshmukh2312) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:883]: Function 'get_cvg_cnt_and_cvg_k8' is not called from anywhere (dead code), so removing it.

**Comment by yatin-mahajan** [comment]
is this PR duplicate/replacement of [PR1605](https://github.com/Seagate/cortx-motr/pull/1605)

**Review by JugalPatil — APPROVED** [review]

**Comment by zahidsk** [comment]
retest this please

**Comment by atulsdeshmukh2312** [comment]
Created custom build with following 
Component | Branch | Repo
Provisioner | CORTX-30761 | https://github.com/Seagate/cortx-prvsnr.git
Motr | CORTX-29834-storage | https://github.com/Seagate/cortx-motr.git
Hare | CORTX-30793 | https://github.com/zahidsk/cortx-hare.git

Final Build with PR:

https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/6088

Successful Deployment with custom build 6088:

https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/4325/

**Comment by mehjoshi** [comment]
image

Alex has failed with the above message. This failure does not look like it is related to the PR changes. Hence, will merge.

**Pull request merged**